services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/DeviceDto.java

```diff
@@ -76,23 +76,19 @@ public DeviceDto() {
      * @param tenantId The id of the tenant.
      * @param deviceId The id of the device.
      * @param device The data of the DTO.
-     * @param autoProvisioned Marks this device as being auto-provisioned.
-     * @param autoProvisioningNotificationSent Marks the auto-provisioning notification for this device as sent.
+     * @param deviceStatus The registry internal status of the device.
      * @param created The instant when the object was created.
      * @param updated The instant of the most recent update.
      * @param version The version of the DTO
      *
      * @return A DTO instance for reading an entry.
      */
     public static DeviceDto forRead(final String tenantId, final String deviceId, final Device device,
-                                    final Boolean autoProvisioned, final Boolean autoProvisioningNotificationSent, final Instant created, final Instant updated, final String version) {
+                                    final DeviceStatus deviceStatus, final Instant created, final Instant updated, final String version) {
```

Comment:
what about *forCreation* and *forUpdate*? It seems odd that one method accepts the DeviceStatus as arg but the others require DeviceStatus' properties explicitly, doesn't it?

Reply:
_forCreation_ only should set the _autoProvisioned_ flag, _forUpdate_ only the _autoProvisioningNotificationSent_ flag.

If we made the method accept a DeviceStatus instance we would invoke the impression to the caller that one can set both flags in those calls which isn't true.

Reply:
I see